### PR TITLE
Add PipelineStoppedException notes to PowerShell API

### LINKED ISSUE #15173: PowerShell.Stop() behaviour causes very different results between PowerShell.Invoke() and PowerShell.InvokeAsync()
## Steps to reproduce

```csharp
[Cmdlet(VerbsLifecycle.Invoke, "Example")]
public class InvokeScriptCommand : PSCmdlet
{
    [Parameter]
    public SwitchParameter Async { get; set; }

    protected override void EndProcessing()
    {
        using (var powershell = PowerShell.Create())
        {
            powershell.AddScript("1;2;3;sleep 10;4;5;6");

            Task<IList<PSObject>> task = null;
            if (Async)
            {
                task = RunAsync(powershell);
            }
            else
            {
                task = Run(powershell);
            }

            Thread.Sleep(2000);

            powershell.Stop();

            foreach (PSObject result in task.GetAwaiter().GetResult())
            {
                Console.WriteLine(result);
            }
        }
    }

    private Task<IList<PSObject>> RunAsync(PowerShell powershell)
    {
            return powershell.InvokeAsync()
                .ContinueWith(psTask =>
                {
                    return (IList<PSObject>)psTask.Result;
                });
    }

    private Task<IList<PSObject>> Run(PowerShell powershell) => Task.Run(() => (IList<PSObject>)powershell.Invoke());
}
```

```powershell
> Invoke-Example -Async
Invoke-Example: One or more errors occurred. (The pipeline has been stopped.)

> Invoke-Example
1
2
3
```

## Expected behavior

Both the async and sync executions should do the same thing.

## Actual behavior

The sync execution returns a partial result and the async execution throws a `PipelineStoppedException`.

~~What's more, when a `PSDataCollection<PSObject>` is passed into the `InvokeAsync()` call, no results are ever inserted and the `DataAdding` and `DataAdded` events are never fired.~~ I passed my collection to the `input` parameter; passing it to the `output` parameter works...

## Environment data

This is with the PowerShell 7.2.0-preview.4 (both as a standalone app with the SDK and as a cmdlet with the exe).

### PR DESCRIPTION
**Pull request by rjmholt**
# PR Summary

Fixes https://github.com/PowerShell/PowerShell/issues/15173.

Adds doc comments about how `PowerShell.Invoke()` and `PowerShell.InvokeAsync()` work with `PowerShell.Stop()`.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/engine/hostifaces/PowerShell.cs [hunk 1]
```diff
@@ -3655,6 +3683,14 @@ private void AppendExceptionToErrorStream(Exception e)
         /// asyncResult object was not created by calling BeginInvoke
         /// on this PowerShell instance.
         /// </exception>
+        /// <exception cref="System.Management.Automation.PipelineStoppedException">
+        /// The running PowerShell pipeline was stopped.
+        /// This occurs when <see cref="PowerShell.Stop"/> or <see cref="PowerShell.StopAsync(AsyncCallback, object)"/> is called.
+        /// To collect partial output in this scenario,
+        /// supply a <see cref="System.Management.Automation.PSDataCollection{T}"> to <see cref="PowerShell.BeginInvoke"/>
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/hostifaces/PowerShell.cs:3690]:
```suggestion
        /// supply a <see cref="System.Management.Automation.PSDataCollection{T}"> to <see cref="PowerShell.BeginInvoke"/> for the <paramref name="output"/> parameter
```

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@rjmholt CodeFactor complains about invalid XML doc. Can you take a look and fix them?

File: src/System.Management.Automation/engine/hostifaces/PowerShell.cs [hunk 2]
```diff
@@ -3222,6 +3234,14 @@ public Task<PSDataCollection<PSObject>> InvokeAsync<T>(PSDataCollection<T> input
         /// <exception cref="ObjectDisposedException">
         /// Object is disposed.
         /// </exception>
+        /// <exception cref="System.Management.Automation.PipelineStoppedException">
+        /// The running PowerShell pipeline was stopped.
+        /// This occurs when <see cref="PowerShell.Stop"/> or <see cref="PowerShell.StopAsync(AsyncCallback, object)"/> is called.
+        /// To collect partial output in this scenario,
+        /// supply a <see cref="System.Management.Automation.PSDataCollection{T}"> for the <paramref name="output"/> parameter,
```
Comment (adityapatwardhan) [inline src/System.Management.Automation/engine/hostifaces/PowerShell.cs:3241]:
```suggestion
        /// supply a <see cref="System.Management.Automation.PSDataCollection{T}" /> for the <paramref name="output"/> parameter,
```

File: src/System.Management.Automation/engine/hostifaces/PowerShell.cs [hunk 3]
```diff
@@ -3284,6 +3304,14 @@ public Task<PSDataCollection<PSObject>> InvokeAsync<T>(PSDataCollection<T> input
...
```
Comment (adityapatwardhan) [inline src/System.Management.Automation/engine/hostifaces/PowerShell.cs:3311]:
```suggestion
        /// supply a <see cref="System.Management.Automation.PSDataCollection{T}" /> for the <paramref name="output"/> parameter,
```

File: src/System.Management.Automation/engine/hostifaces/PowerShell.cs [hunk 4]
```diff
@@ -3655,6 +3683,14 @@ private void AppendExceptionToErrorStream(Exception e)
         /// asyncResult object was not created by calling BeginInvoke
         /// on this PowerShell instance.
         /// </exception>
+        /// <exception cref="System.Management.Automation.PipelineStoppedException">
+        /// The running PowerShell pipeline was stopped.
+        /// This occurs when <see cref="PowerShell.Stop"/> or <see cref="PowerShell.StopAsync(AsyncCallback, object)"/> is called.
+        /// To collect partial output in this scenario,
+        /// supply a <see cref="System.Management.Automation.PSDataCollection{T}"> to <see cref="PowerShell.BeginInvoke"/> for the <paramref name="output"/> parameter
```
Comment (adityapatwardhan) [inline src/System.Management.Automation/engine/hostifaces/PowerShell.cs:3690]:
```suggestion
        /// supply a <see cref="System.Management.Automation.PSDataCollection{T}" /> to <see cref="PowerShell.BeginInvoke"/> for the <paramref name="output"/> parameter
```

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Comment by daxian-dbw** [comment]
@adityapatwardhan Can you review again?

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**